aiohttp/helpers.py

```diff
@@ -1076,3 +1085,17 @@ def status_code_must_be_empty_body(code: int) -> bool:
     """Check if a status code must return an empty body."""
     # https://datatracker.ietf.org/doc/html/rfc9112#section-6.3-2.1
     return code in {204, 304} or 100 <= code < 200
+
+
+def should_remove_content_length(method: str, code: int) -> bool:
+    """Check if a Content-Length header should be removed.
+
+    This should always be a subset of must_be_empty_body
+    """
+    # https://www.rfc-editor.org/rfc/rfc9110.html#section-8.6-7
+    # https://www.rfc-editor.org/rfc/rfc9110.html#section-8.6-8
+    return (
+        code in {204, 304}
```

Comment:
304 should still be dropped from here, right?
If the user sets the header, we should keep it.

Reply:
I'm leaning towards it should still be removed since `Content-Length` is discouraged (SHOULD NOT) for 304 since its not one of `Cache-Control`, `Content-Location`, `Date`, `ETag`, `Expires`, and `Vary`.

If you feel strongly otherwise, I can adjust the behavior to keep it in, I adjust all the tests as well, and add a test for keeping it.

https://datatracker.ietf.org/doc/html/rfc7232#section-4.1

> The server generating a 304 response MUST generate any of the
> following header fields that would have been sent in a 200 (OK)
> response to the same request: Cache-Control, Content-Location, Date,
> ETag, Expires, and Vary.

> Since the goal of a 304 response is to minimize information transfer
> when the recipient already has one or more cached representations, a
> sender SHOULD NOT generate representation metadata other than the
> above listed fields unless said metadata exists for the purpose of
> guiding cache updates (e.g., Last-Modified might be useful if the
> response does not have an ETag field).

Reply:
I'll come back tomorrow to give this a last check, but that sounds reasonable. They should really have linked to that paragraph from the other section that makes it sound as if it should be treated the same as HEAD.